### PR TITLE
TweakXL YAML Editing and Export From TweakBrowserViewModel

### PR DESCRIPTION
**Pull request by dragonzkiller**
Initial implementation of #922. This is a very rough implementation for initial support to generate TweakXL YAML files. This draft is to open discussion on implementation of features and usability of the current method.

Implemented:
- Text editor with highlighting (WIP) for YAML files
- Export entire record from `TweakBrowserViewModel` Records View
- Export single record/flat from `TweakBrowserViewModel` Records View

Not Yet Implemented:
- Exporting Flats from the Flats list (can be exported from Records view)
- `ChunkViewModel` for `ITweakXLItem`
- `!append` and related tags
- Aliases (how would we conceivably do this?)
- Appending to single YAML file vs creating new one on export

How To Use:
1. Create a new project (optional)
2. Open the TweakBrowser
3. Select a Record in the List
  a. Right-click and export from left-hand list (or)
  b. Right-click an item in the right-hand TreeView and select "TweakXL -> Override Value"
4. Save YAML file (default location is "Resources" directory)
5. Open YAML and edit values
6. Save Edits
7. Pack, Install, Run

Topics of Discussion:
- How a user selects records and flats in `TweakBrowserViewModel` (e.g. Step 3 above)
- How an `ITweakXLItem` is structured
  - Properties (current implementation)
  - Only ID, Base, Type, Value fields
- When to read values from TweakDB
  - In the ViewModels (current implementation)
  - In the YAML Parser

### CODE REVIEW

File: WolvenKit.App/Models/TweakXL.cs [hunk 1]
```diff
@@ -228,17 +229,214 @@ public object ReadYaml(IParser parser, Type type)
             return result;
         }
 
+        private void WriteCName(IEmitter emitter, CName cName, string property = "")
+        {
+            if (!cName.Equals(CName.Empty))
+            {
+                if (!property.IsNullOrWhiteSpace())
```
Comment (rfuzzo) [inline WolvenKit.App/Models/TweakXL.cs:236]: Please add braces according to our editor config

File: WolvenKit.App/Models/TweakXL.cs [hunk 2]
```diff
@@ -228,17 +229,214 @@ public object ReadYaml(IParser parser, Type type)
...
+                emitter.Emit(new Scalar(cName.GetResolvedText()));
+            }
+        }
+
+        private void WriteTweakDBID(IEmitter emitter, TweakDBID tweakDBID, string property = "")
+        {
+            if (tweakDBID.Length > 0)
+            {
+                if (!property.IsNullOrWhiteSpace())
+                    emitter.Emit(new Scalar(property));
```
Comment (rfuzzo) [inline WolvenKit.App/Models/TweakXL.cs:248]: Same as above, code style

File: WolvenKit.App/Models/TweakXL.cs [hunk 3]
```diff
@@ -228,17 +229,214 @@ public object ReadYaml(IParser parser, Type type)
...
+
+                emitter.Emit(new Scalar(tweakDBID.GetResolvedText()));
+            }
+        }
+        private void WriteREDRaRef(IEmitter emitter, IRedResourceAsyncReference raRef, string property = "")
+        {
+            if (!property.IsNullOrWhiteSpace())
+                emitter.Emit(new Scalar(property));
```
Comment (rfuzzo) [inline WolvenKit.App/Models/TweakXL.cs:256]: And here, etc

**Review by rfuzzo — COMMENTED** [review]
Thanks! Could you run a code format over it? Are you using VS22 ?

**Comment by dragonzkiller** [comment]
> Thanks! Could you run a code format over it? Are you using VS22 ?

Yes, I'm on VS22. I ran a cleanup across my files that cleaned up a few things

**Review by rfuzzo — APPROVED** [review]

**Pull request merged**